src/bundle/Controller/User/UserForgotPasswordController.php

```diff
@@ -0,0 +1,185 @@
+use EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory;
+use EzSystems\EzPlatformAdminUiBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\Translation\Exception\InvalidArgumentException;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\API\Repository\Values\User\UserTokenUpdateStruct;
+
+class UserForgotPasswordController extends Controller
+{
+    /** @var FormFactory */
+    private $formFactory;
+
+    /** @var UserService */
+    private $userService;
+
+    /** @var \Swift_Mailer */
+    private $mailer;
+
+    /** @var \Twig_Environment */
+    private $twig;
+
+    /** @var string */
+    private $tokenTtl;
+
+    /**
+     * @param FormFactory $formFactory
+     * @param UserService $userService
+     * @param \Swift_Mailer $mailer
+     * @param \Twig_Environment $twig
+     * @param string $tokenTtl
+     */
+    public function __construct(
+        FormFactory $formFactory,
+        UserService $userService,
+        \Swift_Mailer $mailer,
+        \Twig_Environment $twig,
+        string $tokenTtl
+    ) {
+        $this->formFactory = $formFactory;
+        $this->userService = $userService;
+        $this->mailer = $mailer;
+        $this->twig = $twig;
+        $this->tokenTtl = $tokenTtl;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     * @throws InvalidOptionsException
+     * @throws \InvalidArgumentException
+     * @throws InvalidArgumentException
+     * @throws UnauthorizedException
+     * @throws NotFoundException
+     */
+    public function userForgotPasswordAction(Request $request): Response
+    {
+        $form = $this->formFactory->forgotUserPassword();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $users = $this->userService->loadUsersByEmail($data->getEmail());
+
+            if (count($users) > 1) {
+                return $this->redirectToRoute('ezplatform.user.forgot_password.login');
+            }
+
+            if (!empty($users)) {
+                $user = reset($users);
+
+                $token = $this->updateUserToken($user);
+
+                $this->sendResetPasswordMessage($user->email, $token);
+            }
+
+            return $this->render('@EzPlatformAdminUi/Security/forgot_user_password_success.html.twig');
+        }
+
+        return $this->render('@EzPlatformAdminUi/Security/forgot_user_password.html.twig', [
+            'form_forgot_user_password' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Exception
+     * @throws InvalidOptionsException
+     * @throws \InvalidArgumentException
```

Comment:
`\InvalidArgumentException` and `InvalidArgumentException` is very confusing. Can you import kernel exceptions with an alias: `ApiException\InvalidArgumentException`?